clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4903,7 +4903,7 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   }
 
   if (Arg *A = Args.getLastArg(options::OPT_LongDouble_Group)) {
-    if (TC.getTriple().isX86())
+    if (TC.getTriple().isX86() || TC.getTriple().isSPIR())
       A->render(Args, CmdArgs);
     else if ((TC.getArch() == llvm::Triple::ppc || TC.getTriple().isPPC64()) &&
```

Comment:
I can imagine a couple reasons not to merge these `if`s:
1. Conflicting changes from llvm.org would be easier to resolve
2. Better readability of the complicated condition for the second branch
Does any of these stand from your viewpoint, or am I missing any additional logic here?

Reply:
There is definitely the desire to keep these separate due to the specific `-mlong-double-80` condition.  Keeping it simple makes sense to me.

Reply:
Interesting discussion...
But why not stating this in a comment around this to avoid some technical-debt time-bomb? :-)